docs/DeveloperGuide.md

```diff
@@ -1,38 +1,177 @@
+4. Split the string using ` -q ` regardless of whether the string contains the quantity part
+5. Check the index of `-n` and `-e` and split the string to save the details of the food name and expiry date
+6. If `hasQuantity` is `true`, return a String array with `name`, `date` and `quantity`, else only return
+a String array with `name` and `date`.
+7. Create a new `Food` object and use `addFood` to add the new food into the food list
+8. Return a `CommandResult` to show the successful message to the user
 
-{Describe the target user profile}
 
-### Value proposition
+**Class Diagram**
+
+![ClassDiagram](images/AddCommandClassDiagram.png)
+
+
+**Sequence Diagram**
+
+![ClassDiagram](images/AddCommandSequenceDiagram.png)
+
+### Remove feature
+The remove command is implemented using a `RemoveCommand` class which
+utilizes `removeFood` method of the FoodList.
+
+The `removeFood` method implementation is as follows:
+1. New `RemoveCommand` object is created by passing in a String containing arguments from `Parser`.
+2. The `index` of the item to be deleted will be taken as its arguments.
+3. The method `execute` of `Command` class will be called from `Duke` with the `index`.
+4. This method will determine if there exist a food item with that `index` in the `FoodList`.
+5. If it exists, the food item will be deleted and feedback to the user that it has been successfully removed.
+6. Otherwise, the feedback will be an error message
+7. This method will return an object called `CommandResult` and pass `feedbackToUser` as its argument.
+8. `Duke` will subsequently call `printResult` method from `CommandResult`.
 
-{Describe the value proposition: what problem does it solve?}
+Class Diagram (to be added)
+Object Diagram (to be added)
 
-## User Stories
+### Find feature
+The find command is implemented using a `FindCommand` class which
+utilizes `findFood` method of the FoodList.
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+The `findFood` method implementation is as follows:
+1. Initialize `FoodList` result variable
+2. Start a loop iterating over each foodItem 
+3. Do word processing on a term and check if the foodItem name includes it
+4. Enter a loop with switch statement for each component of the `flags` argument 
+5. If some flag is not satisfied, continue the loop without adding foodItem to the result
+6. If the flags loop is over, the flags conditions were satisfied, so add the foodItem to the result
 
-## Non-Functional Requirements
+**Class Diagram** 
+![ClassDiagram](images/FindCommandClassDiagram.png)
 
-{Give non-functional requirements}
 
-## Glossary
+**Sequence Diagram**
+![ClassDiagram](images/FindCommandSequenceDiagram.png)
```

Comment:
I think too many opt blocks are nested each other with impairing the readability. It would be better to simplify the diagram. I have cited the sentence from course website in this comment.
> Keep diagrams simple. The aim is to make diagrams comprehensible, not necessarily comprehensive.